ca/config.go

```diff
@@ -138,27 +139,43 @@ func (s *SecurityConfig) UpdateRootCA(cert, key []byte, certExpiry time.Duration
 	}
 
 	// the RootCA pool should validate against the TLS certificate in the credentials
-	if s.ClientTLSCreds != nil {
-		s.ClientTLSCreds.UpdateCAs(rootCA.Pool, nil)
+	clientConfig := s.ClientTLSCreds.Config()
+	updatedClientConfig, err := NewClientTLSConfig(clientConfig.Certificates, rootCA.Pool, ManagerRole)
```

Comment:
Ah right... :|  in that case I will go with your suggestion of keeping a copy of the TLS config.  I was hoping to do most of the cloning in the `config`.  I actually wonder if we need to take out a lock on `SecurityConfig`, though, as well because `RenewTLSConfigNow` might update everything with an old root CA after `UpdateRootCA` is called.